file_flattener.go

```diff
@@ -48,7 +54,9 @@ func FlattenedFile(originalFile *File) (*File, error) {
 
 	// Merge each original batch into a new batch
 	newBatchesByHeader := map[string][]mergeable{}
-	for _, batch := range originalBatches {
+	for i := range originalBatches {
+		batch := originalBatches[i]
```

Comment:
I've had issues in the past with the loop variable syntax. There are some open issues with Go to fix them, but I tend to discourage their use.